lib/handlers/connect.js

```diff
@@ -14,14 +14,26 @@ function ClientPacketStatus (client, packet) {
 }
 
 var connectActions = [
+  pauseParsing,
   authenticate,
   fetchSubs,
   restoreSubs,
   storeWill,
   doConnack,
+  resumeParsing,
   emptyQueue
 ]
 
+function pauseParsing (arg, done) {
+  this.client._pause()
+  done()
+}
+
+function resumeParsing (arg, done) {
+  this.client._unpause()
+  done()
+}
```

Comment:
These are sync steps, I will try to embed them in others, i.e. authenticate and emptyQueue.